server/users.go

```diff
@@ -172,6 +210,14 @@ func (s *Service) UpdateUser(w http.ResponseWriter, r *http.Request) {
 	if req.Scheme != "" {
 		u.Scheme = req.Scheme
 	}
+	if req.Roles != nil {
```

Comment:
Is there a need for this? You check ValidUpdate above.  If you are trying to set default roles, that should also be done in the validation.  That way there is a single function that takes in data and produces the "correct" format.

Reply:
As far as we're thinking, we still need to set the roles on the user. ValidUpdate only ensures that the structure of the data that we receive is valid.

Reply:
I think ValidUpdate should not only validate the structure but also set defaults if it needs to.